components/layout/fragment.rs

```diff
@@ -1549,66 +1444,39 @@ impl Fragment {
                 let block_flow = info.flow_ref.as_block();
                 result.union_block(&block_flow.base.intrinsic_inline_sizes)
             }
-            SpecificFragmentInfo::Image(ref mut image_fragment_info) => {
-                let mut image_inline_size = match self.style.content_inline_size() {
+            SpecificFragmentInfo::Image(_) | SpecificFragmentInfo::Canvas(_) |
```

Comment:
Good idea, will do. There are so many repeated `is_replaced_or_*` code in `fragment.rs`.